workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -810,5 +810,45 @@ for (var j = 0; j < commandDetailsList.Count; j++) {
                 chunkArray.Dispose();
             }
         }
+
+        internal class AcknowledgeAuthorityLossHandler : AbstractAcknowledgeAuthorityLossHandler
+        {
+            public override EntityArchetypeQuery Query => new EntityArchetypeQuery
+            {
+                All = new ComponentType[]
+                {
+                    ComponentType.ReadOnly<AuthorityLossImminent<<#= componentNamespace #>.Component>>(),
+                    ComponentType.ReadOnly<SpatialEntityId>()
+                },
+                Any = Array.Empty<ComponentType>(),
+                None = Array.Empty<ComponentType>()
+            };
+
+            public override void AcknowledgeAuthorityLoss(ComponentGroup group, ComponentSystemBase system,
+                Improbable.Worker.Core.Connection connection)
+            {
+                var authorityLossType = system.GetArchetypeChunkComponentType<AuthorityLossImminent<<#= componentNamespace #>.Component>>();
+                var spatialEntityType = system.GetArchetypeChunkComponentType<SpatialEntityId>();
+
+                var chunkArray = group.CreateArchetypeChunkArray(Allocator.TempJob);
```

Comment:
Wrap `chunkArray` in a `using` block

Reply:
It's a struct that contains unsafe code. It's safer to do it explicitly rather than dispose a copy.

Reply:
Wow, what a terrible API...

Reply:
The cost of no garbage I suspect :(